packages/react-i18n/src/i18n.ts

```diff
@@ -9,6 +9,7 @@ import {
   TimeUnit,
   Weekdays,
 } from '@shopify/javascript-utilities/dates';
+import {applyTimeZoneOffset} from '@shopify/dates';
```

Comment:
What does this timezone & date related stuff have to do with currency formatting? Is it because of `NODE_ICU_DATA`?

Reply:
I think that adding `full_icu` is what triggered this.
At the end of the day, this test was ill-defined:  it was setting a date using a timestamp in a specific timezone, and then asserting that the day of the week was a certain value in the "local" timezone where the unit test was running.  This would work sometimes, depending on what timezone was configured in the environment where the unit test ran.  (I got to the point where it passed in my local `dev test`, but failed in CI.) . So, yes, this has nothing to do with currency formatting--but the unit test was failing, for a bogus reason, which was blocking my PR from being merged.